### DISCUSSION TITLE
Convertion to ONNX format

### DISCUSSION THREAD

**Discussion by blackHorz** [qubvel/segmentation_models.pytorch#438] [General]
Hi,
I found the implementation very efficient and strainght forward. However I would like to know if you can provide me hint for doing inference with OPENCV. 
So I can convert the model to ONNX, however how should I handle the preprocessing function?
Any small code will suffice.
Thanks

**Comment by qubvel** [comment]
You can embed preprocessing function into your model and then convert to ONNX

```python
class InferenceModel(torch.nn.Module):
    def __init__(self, my_model, mean, std):
        super().__init__()
        self.my_model = my_model
        self.register_buffer("mean", mean)
        self.register_buffer("std", std)
    
    def forward(self, x):
        x = x - self.mean
        x = x / self.std
        x = self.my_model(x)
        return x

```
